realm/realm-library/src/objectServer/java/io/realm/sync/permissions/RealmPrivileges.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.sync.permissions;
+
+import io.realm.RealmModel;
+import io.realm.internal.ManagableObject;
+
+/**
+ * FIXME: Javadoc
+ * Thread confined object representing the current level of access to either the Realm, Class
+ * or individual objects.
+ *
+ */
+public class RealmPrivileges implements ManagableObject {
```

Comment:
this needs a `shared_realm` pointer so we can delegate to the ObjectStore method https://github.com/realm/realm-object-store/commit/c80348445d7e355ba385d2119952020df03d16af#diff-4d344943bb20e4505258929b6f4707f3R318